File: c/remove-item-from-array.c
#include <stdio.h>
int main() {
    int nums[3] = {1, 2, 3};

    printf("[ ");
    int size = sizeof nums / sizeof nums[0];
    for(int i = 0; i < size; i++) {
        printf("%d, ", nums[i]);
    }
    printf(" ]\n");

    char newArr[size-1];
    int j = 0;
    for(int i = 0; i < size; i++) {
        if(nums[i] != 2) {
            newArr[j] = nums[i];
            j++;
        }
    }

    printf("[ ");
    for(int i = 0; i < size-1; i++) {
        printf("%d, ", newArr[i]);
    }
    printf(" ]\n");

    return 0;
}

File: zig/remove-item-from-array.zig
const std = @import("std");

pub fn main() void {
    const nums = [_]i32{ 1, 2, 3 };

    std.debug.print("[", .{});
    for (nums) |num| {
        std.debug.print("{}", .{num});
    }
    std.debug.print("]\n", .{});

    var newArr: [nums.len - 1]i32 = undefined;
    var j: usize = 0;
    for (nums) |num| {
        if (num != 2) {
            newArr[j] = num;
            j += 1;
        }
    }

    std.debug.print("[", .{});
    for (newArr) |num| {
        std.debug.print("{}", .{num});
    }
    std.debug.print("]\n", .{});
}
